app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,35 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @users = User.all
+    @user = User.find(params[:user_id])
+    @posts = @user.return_recent_posts
+  end
 
-  def show; end
+  def show
+    @users = User.all
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:id])
+    @comments = @post.return_recent_comments.each
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new
```

Comment:
- [ ] I would suggest you use this way in creating a post `  @post = @current_user.posts.new()`

Please remember [Strong parameters](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)